examples/relational_algebra.dx

```diff
@@ -0,0 +1,122 @@
+'# The Relational Algebra
+Unlike other languages which have bespoke "table"  or "dataframe" datatypes, in principle
+Dex's record system natively allows the manipulation of tables of records
+to do the sorts of data-munging usually done by packages such as pandas or dplyr.
+
+'This is an attempt to express the basic operations on datasets
+described by the relational algebra,
+roughly following [Wikipedia](https://en.wikipedia.org/wiki/Relational_algebra).
+
+'Right now there are two major differences from the standard relational algebra:
+1. We're not using sets of records, but rather tables.  I'm not sure which is better.
+2. Pretty much all the output types of these functions need to be flattened in order to match their standard defintions, but I don't know how to type this.
+
+'### Cartesian Product
+
+def cartesian_product (r:n=>a) (s:m=>b) : (n & m)=>(a & b) =
+  -- Follows the set theory defintion, not the relational
+  -- algebra definition, which would flatten a & b.
```

Comment:
But... it does flatten to `a & b`

Reply:
I wrote that poorly, I said it "would flatten `a & b`", i.e. the relational algebra definition would flatten `a & b` to `(a_1, a_2, ... b_1, b_2...)`, but this implementation doesn't do that.